Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -0,0 +1,297 @@
+﻿#nullable enable
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Discord;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Modix.Data;
+using Modix.Data.Extensions;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Moderation;
+using Modix.Services.AutoRemoveMessage;
+using Modix.Services.Extensions;
+using Modix.Services.Images;
+using Modix.Services.Utilities;
+
+namespace Modix.Services.UserInfo
+{
+    public class UserInfoCommand : IRequest
+    {
+        public UserInfoCommand(IMessage message, ulong guildId, ulong userId)
+        {
+            Message = message;
+            GuildId = guildId;
+            UserId = userId;
+        }
+        public ulong UserId { get; }
+        public ulong GuildId { get; }
+        public IMessage Message { get; }
+    }
+
+    public class UserInfoCommandHandler : AsyncRequestHandler<UserInfoCommand>
+    {
+        private readonly ModixContext _modixContext;
+        private readonly IDiscordClient _discordClient;
+        private readonly IAutoRemoveMessageService _autoRemoveMessageService;
+        private readonly IImageService _imageService;
+        private readonly ILogger<UserInfoCommandHandler> _logger;
+        private readonly ModixConfig _modixConfig;
+        private readonly Stopwatch _stopwatch = new Stopwatch();
+
+        public UserInfoCommandHandler(ModixContext modixContext, IDiscordClient discordClient,
+            IAutoRemoveMessageService autoRemoveMessageService, IImageService imageService, ILogger<UserInfoCommandHandler> logger,
+            IOptions<ModixConfig> modixOptions)
+        {
+            _modixContext = modixContext;
+            _discordClient = discordClient;
+            _autoRemoveMessageService = autoRemoveMessageService;
+            _imageService = imageService;
+            _logger = logger;
+            _modixConfig = modixOptions.Value;
+        }
+
+        protected override async Task Handle(UserInfoCommand request, CancellationToken cancellationToken)
+        {
+            _stopwatch.Start();
+
+            _logger.LogDebug("Getting info for user {UserId} in guild {GuildId}", request.UserId, request.GuildId);
+
+            var guild = await _discordClient.GetGuildAsync(request.GuildId);
```

Comment:
It looks like this, and the other Discord client methods, take a cancellation token via their `RequestOptions` parameter. Since we have a cancellation token should we be passing it along?

Reply:
We haven't really spoken about cancellation strategy, because there is no where for us to actually cancel the request. I'll pass it in for brevity, but we might need to discuss actual cancellation strategy in another issue.

Reply:
From the bot-frontend, there's no way to cancel something, but there's definitely the possibility to cancel stuff from the Web API, and in theory, we want all of our core logic (not render-related) to be shareable between both.